### PR TITLE
Vertical dynamic type improvements

### PR DESCRIPTION
**Pull request by yuki-stripe**
## Summary
Improve dynamic type support in vertical mode
Also renames PaymentMethodRowButton -> SavedPaymentMethodRowButton

## Testing
For the life of me I can't figure out how to properly override the traitCollection in a snapshot test, so I'm manually testing for now:
Small:

XXLarge:

### Card scan:
| Before  | After |
| ------------- | ------------- |
|CleanShot 2024-06-26 at 12 27 08@2x | CleanShot 2024-06-26 at 12 26 47@2x |
| CleanShot 2024-06-26 at 12 28 50@2x | CleanShot 2024-06-26 at 12 28 26@2x |

### CODE REVIEW

**Review by porter-stripe — DISMISSED** [review]

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Vertical Main Screen/FormHeaderView.swift [hunk 1]
```diff
@@ -39,6 +39,7 @@ final class FormHeaderView: UIView {
         let views = [imageView, label].compactMap { $0 }
         let stackView = UIStackView(arrangedSubviews: views)
         stackView.spacing = 12
+        stackView.alignment = .center
```
Comment (porter-stripe) [inline StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Vertical Main Screen/FormHeaderView.swift:42]: Hmm, why is this needed?

**Review by porter-stripe — DISMISSED** [review]

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Vertical Main Screen/FormHeaderView.swift (hunk 1 above)
Comment (yuki-stripe) [reply StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Vertical Main Screen/FormHeaderView.swift:42]: I believe the default is `.fill`, so the stackview's height ends up being the image's height, with the label shrinking to fit.

**Review by porter-stripe — APPROVED** [review]
